apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/ValueConversion.cs

```diff
@@ -180,24 +175,25 @@ public static Value ToValue(object value, SpannerDbType spannerDbType)
                 case TypeCode.Bool:
                     return new Value {BoolValue = Convert.ToBoolean(value)};
                 case TypeCode.String:
+                    if (value is DateTime)
+                    {
+                        // If the value is a DateTime, we always convert using XmlConvert.
+                        // This allows us to convert back to a datetime reliably from the
+                        // resulting string (so roundtrip works properly if the developer uses
+                        // a string as a backing field for a datetime for whatever reason).
+                        return new Value { StringValue = XmlConvert.ToString((DateTime)value, XmlDateTimeSerializationMode.Utc) };
```

Comment:
I'm using XmlConvert to parse RFC 3339 so I'm using it in both directions for clarity.
I would feel worse about doing say a parseexact and getting a slight variation in the format that XmlConvert might handle.